### DISCUSSION TITLE
How to manage mutation for a list

### DISCUSSION THREAD

**Discussion by yishay-at-bay** [TanStack/query#7573] [Q&A]
Lets say that I have a list of tasks and that in each of them I can check the task as done/undone and save that update to the server with a mutation

Should I create one instance of this mutation for all the tasks? Or should I create an instance of the mutation for each task?
In terms of performance a single instance is better for sure. But are there any other considerations that might make me favor multiple instances?

For example, if I use one instance and a user updates the done checkbox in a number of different tasks in sequence and doesn't wait for a single task update to complete before updating another, can it cause issues in terms of success/error callbacks?

**Comment by TkDodo** [comment]
I'd create one useMutation for each task
